src/core/ext/filters/client_channel/resolver/xds/xds_resolver.h

```diff
@@ -31,23 +31,23 @@ namespace grpc_core {
 class XdsClusterMap;
 class ClusterState;
 
-class XdsClusterLbDataAttribute
+class XdsClusterDataAttribute
     : public ServiceConfigCallData::CallAttributeInterface {
  public:
   static UniqueTypeName TypeName() {
     static UniqueTypeName::Factory factory("xds_cluster_lb_data");
     return factory.Create();
   }
 
-  explicit XdsClusterLbDataAttribute(RefCountedPtr<XdsClusterMap> cluster_map);
+  explicit XdsClusterDataAttribute(RefCountedPtr<XdsClusterMap> cluster_map);
 
-  bool LockClusterConfig(absl::string_view cluster_name);
+  RefCountedPtr<ClusterState> LockAndGetClusterConfig(
```

Comment:
Please document that this method should be called exactly once, at which point we drop our ref to the cluster map, so this attribute will no longer contain anything useful.

Reply:
I believe this is a better code if we split it into `GetClusterState` and `ReleaseClusterMap` - it makes code easier to follow and more logical.

Reply:
I think splitting this into two methods actually makes this API easier to misuse.  There is no case in which we want a caller to lookup a cluster without unreffing the map, so the API shouldn't allow that.

Reply:
Looks like you missed this comment from my last pass.  Please recombine `GetClusterState()` and `ReleaseClusterMap()` into a single method, so that it's not possible for the caller to misuse the API by calling one without the other.

Reply:
I did that. Should the null check be replaced with an assert?